### PR TITLE
add basic arch doc

### PR DESCRIPTION
**Pull request by toddbaert**
* adds high level arch doc
* removes rp-versioning from `/deployment`

### CODE REVIEW

File: web-docs/architecture.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Architecture
+
+flagd architectures fall into two broad categories: those where the evaluation engine is deployed in a standalone process to which the client application connects ([RPC](#rpc-evaluation)), and those where the evaluation engine is embedded into the client application ([in-process](#in-process-evaluation)).
+
+## RPC vs In-Process Evaluation 
+
+### RPC evaluation
+
+In RPC-based deployments one or more flagd instances deployed and exposed to client applications in your infrastructure.
+flagd RPC providers use HTTP or gRPC to request flag evaluations from flagd.
```
Comment (beeme1mr) [inline web-docs/architecture.md:10]: Have we explained what a provider is by this point of the docs? Either way, we may want to create a glossary section with terms like provider.

https://squidfunk.github.io/mkdocs-material/reference/tooltips/?h=glo#adding-a-glossary-includesabbreviationsmd

File: web-docs/architecture.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+The request payload contains the [flag key](https://openfeature.dev/specification/glossary#flag-key) identifying the flag to be evaluated, as well as the relevant [evaluation context](https://openfeature.dev/specification/glossary#evaluation-context).
+The flagd instance is configured to watch one or many [syncs](./concepts/syncs.md), and merges them to build its set of flags (see [here](./concepts/syncs.md#merging) for more details on flag definition merging).
```
Comment (beeme1mr) [inline web-docs/architecture.md:12]:
```suggestion
The flagd instance is configured to watch one or more [syncs](./concepts/syncs.md), and merges them to build its set of flags (see [here](./concepts/syncs.md#merging) for more details on flag definition merging).
```

File: web-docs/architecture.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+When sync sources are updated, flagd will send notifications to clients that flags have changed, enabling applications to react to changes by re-evaluating flags.
+
+This architecture is can be leveraged by very simple clients, since no in-process engine is needed; in fact, you can evaluate flags directly from a terminal console using the `cURL` utility.
+One disadvantage of this pattern is the latency involved in the remote request (though flagd takes ~10ms for an evaluation, and can evaluate thousands of flags per second).
```
Comment (beeme1mr) [inline web-docs/architecture.md:16]: 10ms is higher than I've experienced. I typically see ~2ms, but that depends on multiple factors.

**Review by beeme1mr — APPROVED** [review]

File: web-docs/architecture.md (hunk 1 above)
Comment (toddbaert) [reply web-docs/architecture.md:10]: I think we should use the glossary in openfeature.dev for OpenFeature terms, but I would be OK with adding a glossary for flagd specific stuff.

File: web-docs/architecture.md (hunk 3 above)
Comment (toddbaert) [reply web-docs/architecture.md:16]: We could just say "sub 10ms".

File: web-docs/architecture.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+In RPC-based deployments one or more flagd instances deployed and exposed to client applications in your infrastructure.
+flagd RPC providers use HTTP or gRPC to request flag evaluations from flagd.
+The request payload contains the [flag key](https://openfeature.dev/specification/glossary#flag-key) identifying the flag to be evaluated, as well as the relevant [evaluation context](https://openfeature.dev/specification/glossary#evaluation-context).
+The flagd instance is configured to watch one or more [syncs](./concepts/syncs.md), and merges them to build its set of flags (see [here](./concepts/syncs.md#merging) for more details on flag definition merging).
+When sync sources are updated, flagd will send notifications to clients that flags have changed, enabling applications to react to changes by re-evaluating flags.
+
+This architecture is can be leveraged by very simple clients, since no in-process engine is needed; in fact, you can evaluate flags directly from a terminal console using the `cURL` utility.
+One disadvantage of this pattern is the latency involved in the remote request (though flagd takes ~10ms for an evaluation, and can evaluate thousands of flags per second).
```
Comment (toddbaert) [inline web-docs/architecture.md:16]:
```suggestion
One disadvantage of this pattern is the latency involved in the remote request (though flagd typically takes  <10ms for an evaluation, and can evaluate thousands of flags per second).
```

**Pull request merged**